### PR TITLE
feat(pt): add custom save behaviors

### PR DESCRIPTION
**Pull request by OutisLi**
## Summary by CodeRabbit

* **New Features**
  * Added `training.save_dir` for periodic checkpoints and `validating.save_best_dir` for best-validation checkpoints.
  * Added `training.ckpt_keep_ratio` for ratio-based sliding-window checkpoint retention.

* **Bug Fixes**
  * Improved checkpoint filename/“latest” aliasing and symlink/pointer behavior for periodic and EMA checkpoints when `save_dir` is set.
  * Ensured “best” checkpoints are written only to the configured best-checkpoint directory.
  * Eagerly creates the validator checkpoint directory during initialization.

* **Documentation**
  * Documented `save_dir` and `ckpt_keep_ratio`; updated the training advanced guide and example config.

* **Tests**
  * Added unit tests for retention rounding/edge cases and filesystem tests for redirecting checkpoints and custom best-checkpoint locations.

### CODE REVIEW

**Comment by wanghan-iapcm** [comment]
Non-blocking doc nit: the documented `ckpt_keep_ratio` formula doesn't match the code.

Both the argcheck help text and the user doc state a single ceil:

> `ceil(ckpt_keep_ratio * numb_steps / save_freq)`

but `resolve_keep_ckpt_count` computes a **nested** ceil — `ceil(ratio * ceil(num_steps / save_freq))`:

https://github.com/deepmodeling/deepmd-kit/blob/b0a4b15974443cdefab0986c4a0f1c98479fdb49/deepmd/pt/train/utils.py#L283-L285

These differ whenever `numb_steps` isn't a multiple of `save_freq`. E.g. `numb_steps=11, save_freq=10, ratio=0.9`: the code keeps **2**, the documented formula gives **1**. The code is correct; only the docs are off.

Suggest updating both to the nested form `ceil(ckpt_keep_ratio * ceil(numb_steps / save_freq))` (the `resolve_keep_ckpt_count` docstring already states it correctly):

https://github.com/deepmodeling/deepmd-kit/blob/b0a4b15974443cdefab0986c4a0f1c98479fdb49/deepmd/utils/argcheck.py#L4997-L5002

https://github.com/deepmodeling/deepmd-kit/blob/b0a4b15974443cdefab0986c4a0f1c98479fdb49/doc/train/training-advanced.md#L107-L107

**Review by wanghan-iapcm — COMMENTED** [review]
see my non-blocking comment

**Comment by OutisLi** [comment]
> see my non-blocking comment

done

File: examples/water/dpa4/input.json [hunk 1]
```diff
@@ -119,6 +120,7 @@
   },
   "validating": {
     "compiled_infer": false,
-    "tf32_infer": false
+    "tf32_infer": false,
+    "save_best_dir": "ckpt_best"
```
[the remark this answers is not in the document]
Comment (OutisLi) [reply examples/water/dpa4/input.json:124]: just here to show this key

**Review by wanghan-iapcm — APPROVED** [review]
